packages/assets-controllers/src/assetsUtil.test.ts

```diff
@@ -436,4 +439,134 @@ describe('assetsUtil', () => {
       expect(timestampsIncreasing).toBe(true);
     });
   });
+
+  describe('fetchAndMapExchangeRates', () => {
+    const mockedPricesService: AbstractTokenPricesService = {
+      validateChainIdSupported(_chainId: unknown): _chainId is Hex {
+        return true;
+      },
+      validateCurrencySupported(_currency: unknown): _currency is string {
+        return true;
+      },
+      async fetchTokenPrices() {
+        return {};
+      },
+    };
+
+    it('should return empty object when chainId not supported', async () => {
+      const testTokenAddress = '0x7BEF710a5759d197EC0Bf621c3Df802C2D60D848';
+
+      jest
+        .spyOn(mockedPricesService, 'validateChainIdSupported')
+        .mockReturnValue(false);
+
+      const result = await assetsUtil.fetchTokenContractExchangeRates({
+        tokenPricesService: mockedPricesService,
+        nativeCurrency: 'ETH',
+        tokenAddresses: [testTokenAddress],
+        chainId: '0x1',
```

Comment:
Nit: just so it's clear that we're using a mock, unsupported chainId.
```suggestion
        chainId: '0x0',
```